privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -654,6 +654,60 @@ def testconnection(cls, param):
         
         return success, desc
 
+    def add_user(self, uid, object_class=None, attributes=None):
+        """
+        Add a new user to the LDAP directory.
+
+        attributes are these
+        "username", "surname", "givenname", "email",
+        "mobile", "phone", "password"
+
+        :param uid: The uid of the user object in the resolver
+        :type uid: string
+        :param object_class: Attributes according to the attribute mapping
+        :type object_class: list
+        :param attributes: Attributes according to the attribute mapping
+        :type attributes: dict
+        :return: The new UID of the user. The UserIdResolver needs to
+        determine the way how to create the UID.
+        """
+        attributes = attributes or {}
+        try:
+            self._bind()
+
+            params = self._attributes_to_ldap_attributes(attributes)
+            self.l.add(uid, object_class, params)
```

Comment:
This has implication for the ldap3mock too, the modify method currently only supports dn.
How do you want to handle the changes?
Looks like the following changes would be required:
1. Add additional test(s) to handle add/delete/modify of users with alternative uidtypes
2. Fix the ldap3mock

Does this sound like it could be added to this pull request or would you prefer it separating out?

Best Regards

Martin